master/internal/api_runs.go

```diff
@@ -239,3 +249,155 @@ func sortRuns(sortString *string, runQuery *bun.SelectQuery) error {
+		Join("JOIN projects p ON r.project_id = p.id").
+		Join("JOIN workspaces w ON p.workspace_id = w.id").
+		Where("r.project_id = ?", req.SourceProjectId)
+
+	if req.Filter == nil {
+		getQ = getQ.Where("r.id IN (?)", bun.In(req.RunIds))
+	} else {
+		var efr experimentFilterRoot
+		err := json.Unmarshal([]byte(*req.Filter), &efr)
+		if err != nil {
+			return nil, err
+		}
+		getQ = getQ.WhereGroup(" AND ", func(q *bun.SelectQuery) *bun.SelectQuery {
+			_, err = efr.toSQL(q)
+			return q
+		}).WhereGroup(" AND ", func(q *bun.SelectQuery) *bun.SelectQuery {
+			if !efr.ShowArchived {
+				return q.Where(`e.archived = false`)
+			}
+			return q
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if getQ, err = experiment.AuthZProvider.Get().FilterExperimentsQuery(ctx, *curUser, nil, getQ,
+		[]rbacv1.PermissionType{
+			rbacv1.PermissionType_PERMISSION_TYPE_VIEW_EXPERIMENT_METADATA,
+			rbacv1.PermissionType_PERMISSION_TYPE_DELETE_EXPERIMENT,
+		}); err != nil {
+		return nil, err
+	}
+
+	err = getQ.Scan(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var results []*apiv1.RunActionResult
+	visibleIDs := set.New[int32]()
+	var validIDs []int32
+	// associated experiments to move
+	var expMoveIds []int32
+	for _, check := range runChecks {
+		visibleIDs.Insert(check.ID)
+		if check.Archived {
+			results = append(results, &apiv1.RunActionResult{
+				Error: "Run is archived.",
+				Id:    check.ID,
+			})
+			continue
+		}
+		if check.IsMultitrial && req.SkipMultitrial {
+			results = append(results, &apiv1.RunActionResult{
+				Error: fmt.Sprintf("Skipping run '%d' (part of multi-trial).", check.ID),
+				Id:    check.ID,
+			})
+			continue
+		}
+		if check.ExpID != nil {
+			expMoveIds = append(expMoveIds, *check.ExpID)
+		}
+		validIDs = append(validIDs, check.ID)
+	}
+	if req.Filter == nil {
+		for _, originalID := range req.RunIds {
+			if !visibleIDs.Contains(originalID) {
+				results = append(results, &apiv1.RunActionResult{
+					Error: fmt.Sprintf("Run with id '%d' not found in project with id '%d'", originalID, req.SourceProjectId),
+					Id:    originalID,
+				})
+			}
+		}
+	}
+	if len(validIDs) > 0 {
+		err = db.Bun().RunInTx(ctx, nil, func(ctx context.Context, tx bun.Tx) error {
+			expMoveResults, err := experiment.MoveExperiments(ctx, expMoveIds, nil, req.DestinationProjectId)
+			if err != nil {
+				return err
+			}
+			failedExpMoveIds := []int32{-1}
+			for _, res := range expMoveResults {
+				if res.Error != nil {
+					failedExpMoveIds = append(failedExpMoveIds, res.ID)
+				}
+			}
+			var acceptedIDs []int32
+			if _, err = tx.NewUpdate().Table("runs").
+				Set("project_id = ?", req.DestinationProjectId).
+				Where("runs.id IN (?)", bun.In(validIDs)).
+				Where("runs.experiment_id NOT IN (?)", bun.In(failedExpMoveIds)).
+				Returning("runs.id").
+				Model(&acceptedIDs).
+				Exec(ctx); err != nil {
+				return fmt.Errorf("updating run's project IDs: %w", err)
+			}
```

Comment:
experiment.MoveExperiments should already set this so I don't think we need this update

https://github.com/determined-ai/determined/blob/bc8c31cab990c3c1050ee698a1d09f6ac1328aff/master/internal/experiment/bulk_action.go#L721

We might still need a way to map expMoveResult to the run API response but we might be able to use runChecks to convert ```expMoveResult``` to run IDs

Reply:
added a query to populate move results with failed experiment moves